core/src/main/java/org/apache/iceberg/MetricsConfig.java

```diff
@@ -212,6 +212,26 @@ public void validateReferencedColumns(Schema schema) {
     }
   }
 
+  /**
+   * Validate that all referenced columns in the properties are valid columns in the schema.
+   *
+   * @param schema schema to validate against
+   * @param props properties may contain metrics mode column overrides
+   */
+  public static void validateReferencedColumns(Schema schema, Map<String, String> props) {
```

Comment:
Do we need to make this public? 


Also for implementation could we do 
from(props, null, null).validateReferencedColumns(schema)

Just looking to reduce our duplicated code here if possible, feel free to skip if that doesn't make sense

Reply:
Make sense.

The method is contracted to a non-public. I also reuse the `from(props, null, null)`.
@RussellSpitzer  PTAL.